### DISCUSSION TITLE
Editing list item or better solution ideas for multi select.  EDIT: ANSWERED

### DISCUSSION THREAD

**Discussion by croyleje** [charmbracelet/bubbletea#364] [Q&A] [answered]
I have rewritten a cli application of mine to take advantage of the bubbletea framework and to learn it.  I am having an issue implementing the multi selection of list items.  I found a few examples and been able to make a simple list with multi selection but if using the lists bubble and a struct if I add a bool value to my struct I can't work out a way to toggle it from the list view.  I have added some code snippets below if anyone has a better idea I am all ears.

[link to my repo](https://github.com/croyleje/gorm.git)

Basically I make a struct.
```
type Item struct {
     Name         string
     Type         string
     Size         string
     Path         string
     DeletionDate string
     IsChecked    bool
}
```

Then I have an initialModel.
```
func InitialModel() Model {
     entries, err := cmd.GetEntries()
     if err != nil {
         log.Fatalf("error: %v", err)
     }

     items := []list.Item{}
     for _, e := range entries {
         items = append(items, item{
             Name:         e.Name,
             Type:         e.Type,
             Size:         e.Size,
             Path:         e.Path,
             DeletionDate: e.DeletionDate,
             IsChecked:    e.IsChecked,
         })
     }
```

All of this works fine and so does my views and if I manually go and hard code the `IsChecked` to true the rendering and updates work fine.

But how could I change the `IsChecked` from the a main view if I try to do it from the listUpdate function I have a problem getting access to the item itself.
I have tried multiple different ways below is a couple in order of the way it seems like it should work to me.  Again if anyone could just point me in the correct direction.

Ways tried.

From listUpdate func.
```
case key.Matches(msg, m.keyMap.Select):
     m.list.SelectedItem().(item).IsChecked = true  **ERROR can not assign value of type bool
     m.list.SelectedItem().IsChecked = true  **ERROR list.Item has no field IsChecked
```

I also tried passing the item to a func and doing the same things basically and got basically the same errors.

Next idea and the way I am guessing it is done.  Again via the listUpdate func.
Iterate over the items till I find the one that matches the m.list.SelectedItem().(item) and then append IsChecked: true.  But then I have the issue of this not being saved anywhere and when you move the cursor up or down the list the state is lost.

I know I am missing something and its probably obvious but any pointers in the write direction or a link to something similar would be greatly appreciated.

EDIT ANSWERED:
The `m.list.SetItem` is the answer you can use the `m.list.SelectedItem().(item)` and `m.list.Index()` as your arguments to it and edit the item from there.

Sorry for the trouble.

**Comment by bashbunni** [comment, marked as answer]
Glad you were able to figure it out! If there's anything that could be clarified, please don't hesitate to let us know
